Lab-5/ejemyr_lab5.ipynb

```diff
@@ -0,0 +1,510 @@
+    "\n",
+    "The eqplicit solution of a 2D-polynomial on the reference triangle is given by\n",
+    "\n",
+    "\\begin{equation}\n",
+    "    \\int_0^1 \\int_0^{1-x} c_1x^2 + c_2y^2 + c_3xy + c_4x + c_5y + c_6 \\ dydx = \\frac{c_1}{12} + \\frac{c_2}{12} + \\frac{c_3}{24} + \\frac{c_4}{6} + \\frac{c_5}{6} + \\frac{c_6}{2}\n",
+    "\\end{equation}"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 35,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "def three_point_quad_on_ref_trig(f: Callable[[float, float], float]) -> float:\n",
+    "    w = 1/6\n",
+    "    p0 = (0.5, 0)\n",
+    "    p1 = (0, 0.5)\n",
+    "    p2 = (0.5, 0.5)\n",
+    "    \n",
+    "    return w * f(p0[0], p0[1]) + w * f(p1[0], p1[1]) + w * f(p2[0], p2[1])"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 36,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "class Tests(Tests):\n",
+    "    def test_three_point_quad(self):\n",
+    "        min_length = 1\n",
+    "        max_length = 100\n",
+    "        def genetator():\n",
+    "            c = np.random.rand(6)\n",
+    "            f: Callable[[float, float], float] = lambda x, y: c[0] * x**2 + c[1] * y**2 + c[2]* x*y + c[3]* x + c[4] * y + c[5]\n",
+    "            return three_point_quad_on_ref_trig(f), c[0]/12 + c[1]/12 + c[2]/24 + c[3]/6 + c[4]/6 + c[5]/2\n",
+    "\n",
+    "        Tests.check_accuracy_multiple_random(1000, genetator, 7)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### Monte Carlo quadrature"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Skriv nått"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 50,
+   "metadata": {},
+   "outputs": [
+    {
+     "data": {
+      "text/plain": [
+       "1.3093473924718282"
+      ]
+     },
+     "execution_count": 50,
+     "metadata": {},
+     "output_type": "execute_result"
+    }
+   ],
+   "source": [
+    "def monte_carlo_on_unit_interval(f: Callable[[np.ndarray], np.ndarray], n: int) -> float:\n",
+    "    return f(np.random.rand(n)).mean()\n",
+    "\n",
+    "def monte_carlo_on_ref_trig(f: Callable[[np.ndarray, np.ndarray], np.ndarray], n: int) -> float:\n",
+    "    x = np.random.rand(n)\n",
+    "    x_temp = x.copy()\n",
+    "    y = np.random.rand(n)\n",
+    "    args = np.argwhere((1 - x) < y)\n",
+    "    x.put(args, 1 - y[args])\n",
+    "    y.put(args, 1 - x_temp[args])\n",
+    "    \n",
+    "    return f(x, y).mean(), x, y\n",
+    "\n",
+    "\n",
+    "c = np.random.rand(6)\n",
+    "f = lambda x, y: c[0] * (x**2) + c[1] * (y**2) + c[2]* (x*y) + c[3]* x + c[4] * y + c[5]\n",
+    "true = c[0]/12 + c[1]/12 + c[2]/24 + c[3]/6 + c[4]/6 + c[5]/2\n",
+    "monte_carlo_on_ref_trig(f, 1000)[0]"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {
+    "colab_type": "text",
+    "id": "SsQLT38gVbn_"
+   },
+   "source": [
+    "# **Results**"
```

Comment:
You are missing results for Monte Carlo over the reference triangle.

Reply:
Whops...

Reply:
Fixed it now 👍